database.sql

```diff
@@ -35,32 +35,39 @@ CREATE TABLE IF NOT EXISTS `users` (
 ) ENGINE = InnoDB;
 
 CREATE TABLE IF NOT EXISTS `characters` (
-  `charid` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
-  `userid` int(11) UNSIGNED NOT NULL,
+  `charid` SMALLINT UNSIGNED NOT NULL AUTO_INCREMENT,
+  `userid` SMALLINT UNSIGNED NOT NULL,
   `firstname` varchar(50) NOT NULL,
   `lastname` varchar(50) NOT NULL,
-  `gender` varchar(50) NOT NULL,
+  `gender` varchar(10) NOT NULL,
   `dateofbirth` date NOT NULL,
   `phone_number` varchar(20) DEFAULT NULL,
-  `last_played` date NOT NULL DEFAULT curdate(),
-  `is_dead` tinyint(1) NOT NULL DEFAULT 0,
+  `last_played` date NOT NULL DEFAULT (curdate()),
+  `is_dead` tinyint UNSIGNED NOT NULL DEFAULT 0,
   `x` float DEFAULT NULL,
   `y` float DEFAULT NULL,
   `z` float DEFAULT NULL,
   `heading` float DEFAULT NULL,
-  `inventory` longtext NOT NULL DEFAULT '[]',
-  `metadata` JSON DEFAULT '{}' CHECK (JSON_VALID(`metadata`)),
+  `metadata` JSON DEFAULT (JSON_OBJECT()) CHECK (JSON_VALID(`metadata`)),
   `deleted` DATE NULL DEFAULT NULL,
   PRIMARY KEY (`charid`) USING BTREE,
   KEY `FK_character_users` (`userid`) USING BTREE,
   CONSTRAINT `FK_characters_users` FOREIGN KEY (`userid`) REFERENCES `users` (`userid`) ON UPDATE CASCADE ON DELETE CASCADE
 ) ENGINE = InnoDB;
 
+CREATE TABLE IF NOT EXISTS `characters_inventory` (
+  `charid` SMALLINT UNSIGNED NOT NULL,
+  `inventory` longtext NOT NULL DEFAULT ('[]'),
+  PRIMARY KEY (`charid`),
+  KEY `FK_inventory_characters` (`charid`),
+  CONSTRAINT `FK_inventory_characters` FOREIGN KEY (`charid`) REFERENCES `characters` (`charid`) ON UPDATE CASCADE ON DELETE CASCADE
+) ENGINE = InnoDB;
+
```

Comment:
How much of a difference does this really make for writes and selects..? Pretty sure MySQL 8 doesn't support default values on longtext.

Reply:
> How much of a difference does this really make for writes and selects..?
Well characters table is used often (every character selection, updating metadata (statuses in the future). And even if the inventory is not explicitly selected via query, you have to skip this data, which means that it still has to be read.

> Pretty sure MySQL 8 doesn't support default values on longtext
It does, you can have a default string value.. but it conflicts with [] and {}... that the reason why the default value is in parentheses